openshift/Dockerfile.requirements

```diff
@@ -0,0 +1,59 @@
+FROM registry.ci.openshift.org/ocp/4.17:base-rhel9
+
+RUN yum install -y libffi-devel python3-devel gcc python3-pip
+    
+COPY ./Pipfile* ./
+
+# The build dependencies are required by cachito. Following script
+# does exactly the same. More info at: https://github.com/containerbuildsystem/cachito/blob/master/docs/pip.md#build-dependencies
+RUN curl -LO https://raw.githubusercontent.com/containerbuildsystem/cachito/master/bin/pip_find_builddeps.py \
+    && chmod +x pip_find_builddeps.py
+
+# Create requirements.in file from the pipenv created using the
+# same Pipfile and Pipfile.lock used for upstream image. Then
+# use pip-compile to generate the requirements.txt file. Copy
+# setuptools into requirements-build.txt as pip-compile will
+# omit it from the requierements.txt file.
+RUN python3 -m pip install pipenv==2023.11.15 \
+    && python3 -m pip install pip-tools \
+    && pipenv install --deploy \
+    && pipenv run pip freeze --all  > ./requirements.in \
```

Comment:
pipenv requirements or pipenv lock -r should help to generate better dependency package list compared to freeze.

Reply:
I tried the following commands:
```shell
# pipenv run pip freeze --all > requirements1.txt
# pipenv requirements --exclude-markers > requirements2.txt
# diff -y requirements1.txt requirements2.txt 
							      >	-i https://pypi.org/simple
ansible-core==2.15.11						ansible-core==2.15.11
ansible-runner==2.3.6						ansible-runner==2.3.6
ansible-runner-http==1.0.0					ansible-runner-http==1.0.0
cachetools==5.3.3						cachetools==5.3.3
certifi==2024.2.2						certifi==2024.2.2
cffi==1.16.0							cffi==1.16.0
charset-normalizer==3.3.2					charset-normalizer==3.3.2
cryptography==42.0.7						cryptography==42.0.7
docutils==0.21.2						docutils==0.21.2
google-auth==2.29.0						google-auth==2.29.0
idna==3.7							idna==3.7
importlib-metadata==6.2.1					importlib-metadata==6.2.1
importlib-resources==5.0.7					importlib-resources==5.0.7
Jinja2==3.1.4						      |	jinja2==3.1.4
kubernetes==29.0.0						kubernetes==29.0.0
lockfile==0.12.2						lockfile==0.12.2
MarkupSafe==2.1.5					      |	markupsafe==2.1.5
oauthlib==3.2.2							oauthlib==3.2.2
packaging==24.0							packaging==24.0
pexpect==4.9.0							pexpect==4.9.0
pip==24.2						      <
ptyprocess==0.7.0						ptyprocess==0.7.0
pyasn1==0.6.0							pyasn1==0.6.0
pyasn1_modules==0.4.0					      |	pyasn1-modules==0.4.0
pycparser==2.22							pycparser==2.22
python-daemon==3.0.1						python-daemon==3.0.1
python-dateutil==2.9.0.post0					python-dateutil==2.9.0.post0
PyYAML==6.0.1						      |	pyyaml==6.0.1
requests==2.31.0						requests==2.31.0
requests-oauthlib==2.0.0					requests-oauthlib==2.0.0
requests-unixsocket==0.3.0					requests-unixsocket==0.3.0
resolvelib==1.0.1						resolvelib==1.0.1
rsa==4.9							rsa==4.9
setuptools==69.5.1						setuptools==69.5.1
six==1.16.0							six==1.16.0
urllib3==1.26.18						urllib3==1.26.18
websocket-client==1.8.0						websocket-client==1.8.0
wheel==0.44.0						      <
zipp==3.18.1							zipp==3.18.1
```
As you can see `pipenv requirements` doesn't list `wheel` as a dependency.

Also, `pipenv lock` is used to generate the `Pipfile.lock` file from the `Pipfile`. I am anyway copying both of them so that they are in-sync with what we have upstream: https://github.com/openshift/ansible-operator-plugins/pull/20/files#diff-7ad0e6b82f775968960444bb0baafc660c603428d601e19fdcf45a81623775eeR5

Reply:
We cannot use a single requirements file in OSBS environment for all the packages required by the ansible-operator. The build dependencies are needed to be installed first. Without following that order the build will fail.